components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.ts

```diff
@@ -1,78 +1,69 @@
 import {
-  Component, EventEmitter, OnInit, Input, Output
+  Component, EventEmitter, Input, Output, OnChanges
 } from '@angular/core';
 
-import { orderBy } from 'lodash/fp';
-
-import { Project, ProjectConstants } from 'app/entities/projects/project.model';
+import { ProjectConstants, Project } from 'app/entities/projects/project.model';
 
 const { UNASSIGNED_PROJECT_ID } = ProjectConstants;
 
+export interface ProjectChecked extends Project {
+  checked: boolean;
+}
+
+export function projectCheckedFromProject(project: Project, checked: boolean): ProjectChecked {
+  const projectChecked = <ProjectChecked>project;
+  projectChecked.checked = checked;
+  return projectChecked;
+}
+
 @Component({
   selector: 'app-projects-dropdown',
   templateUrl: './projects-dropdown.component.html',
   styleUrls: ['./projects-dropdown.component.scss']
 })
-export class ProjectsDropdownComponent implements OnInit {
-
-  // The array of projects that you pass into the component. This array gets updated when a user
-  // makes changes and is emitted to the parent component.
-  @Input() projects:  Array<Project> = [];
-
-  // Setting required to true means the dropdown will show an error if a user tries to close the
-  // dropdown but hasn't selected any projects.
-  @Input() required = false;
+export class ProjectsDropdownComponent implements OnChanges {
+  // The map of ProjectChecked by id. Any checked changes propogated via
+  // onProjectChecked. Updates should be applied to parent component state.
+  @Input() projects: { [id: string]: ProjectChecked } = {};
 
   // Setting disabled to true means the dropdown will be unusable and will have a grey background
   @Input() disabled = false;
 
-  // Emits the array of projects to the parent component
-  @Output() onSelection = new EventEmitter<Array<Project>>();
+  // Emits a project that changed as a result of a check or uncheck.
+  @Output() onProjectChecked = new EventEmitter<ProjectChecked>();
 
   active = false;
-  showError = false;
   label = UNASSIGNED_PROJECT_ID;
 
-  // Map of projects currently selected
-  selectedProjects = {};
+  constructor() { }
 
-  constructor() {
-    this.projects = this.sortProjectsByName(this.projects);
+  projectsArray(): ProjectChecked[] {
+    return Object.values(this.projects) ? Object.values(this.projects) : [];
```

Comment:
I don't get it. What is `Object.values(this.projects)` when it's falsy? Did you mean `return this.projects ? Object.values(this.projects) : [];`?

Since this is the constructor, and the default is `{}`, can we just do
```suggestion
    return Object.values(this.projects);
```
?